tests/libs/eavmlib/test_mdns.erl

```diff
@@ -0,0 +1,108 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2025 Paul Guyot <pguyot@kallisys.net>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(test_mdns).
+-include_lib("eunit/include/eunit.hrl").
+
+-define(DNS_TYPE_A, 1).
+-define(DNS_CLASS_IN, 1).
+-define(DNS_OPCODE_STANDARD_QUERY, 0).
+-define(DNS_QR_QUERY, 0).
+-define(DNS_QR_REPLY, 1).
+-define(MDNS_SEND_BROADCAST_RESPONSE, 0).
+-define(MDNS_SEND_UNICAST_RESPONSE, 1).
+
+parse_dns_name_test_() ->
+    [
+        ?_assertEqual(
+            {ok, {[<<"atomvm">>, <<"local">>], <<>>}},
+            mdns:parse_dns_name(<<6, "atomvm", 5, "local", 0>>, <<6, "atomvm", 5, "local", 0>>)
+        ),
+        ?_assertEqual({ok, {[], <<42>>}}, mdns:parse_dns_name(<<0, 42>>, <<0, 42>>)),
+        ?_assertEqual({error, {invalid_name, <<42>>}}, mdns:parse_dns_name(<<42>>, <<42>>)),
+        ?_assertEqual({error, {invalid_name, <<>>}}, mdns:parse_dns_name(<<1, 42>>, <<1, 42>>))
+    ].
+
+parse_dns_message_test_() ->
```

Comment:
I think we should do a couple of tests also for invalid dns messages. Such as a truncated one and a corrupted one. Whatever they way we want to fail, we need to make sure that we'll do in a consistent and reasonable way.